akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -37,6 +39,143 @@ private[stream] object AbstractStage {
   final val NoTerminationPending = 0x8000
   final val BothBalls = UpstreamBall | DownstreamBall
   final val BothBallsAndNoTerminationPending = UpstreamBall | DownstreamBall | NoTerminationPending
+
+  private class PushPullGraphLogic[In, Out, Ext](
```

Comment:
This is the class which adapts a PushPullStage to a GraphStageLogic. Basically an emulation layer on top of the new GraphStage infrastructure.